src/shared/pages/welcomePage/localization.js

```diff
@@ -4,23 +4,30 @@ export function replaceMessage(element, l10nID, href) {
     return false;
   }
 
-  message = message.replace("<a>", `<a id="${l10nID}Link" href="">`);
-
-  // eslint-disable-next-line no-unsanitized/property
-  element.innerHTML = message;
+  if (message.includes("<a>")) {
+    const prefix = message.split("<a>")[0];
+    const suffix = message.split("</a>")[1];
+    const anchor = document.createElement("a");
+    anchor.id = `${l10nID}Link`;
+    anchor.href = href || "";
+    anchor.textContent = message.split("<a>")[1].split("</a>")[0];
+    element.replaceChildren(prefix, anchor, suffix);
+  } else {
+    element.replaceChildren(message);
+  }
 
-  // add a listener to the link if required
-  if (href) {
+  // handle browser shortcut page cases
+  if (l10nID === "welcomePageManageShortcutsFirefox") {
+    const link = document.getElementById(`${l10nID}Link`);
```

Comment:
This logic only works when `element` is part of the document. Otherwise the link you created above cannot be found. The following would be a more reliable version of your intended logic:


```suggestion
    const link = element.querySelector(`#${l10nID}Link`);
```

You don't need that ID though; `a` as selector would also work.